src/blockdev.rs

```diff
@@ -260,6 +280,51 @@ pub fn udev_settle() -> Result<()> {
     Ok(())
 }
 
+/// A Reader of a disk image that fails if the formatted sector size of the
+/// image doesn't match a specified value.
+pub struct DiskSectorSizeReader<R: Read> {
+    source: BufReader<R>,
+}
+
+impl<R: Read> DiskSectorSizeReader<R> {
```

Comment:
We're already reading and keeping the first 1M of the source image, right? Couldn't we just check that buffer? Doing it in a wrapping reader is fine, though it seems like unnecessary cognitive load/architecturing.

Reply:
Good call.  That's much, much better.